phpBB/includes/acp/acp_storage.php

```diff
@@ -326,14 +568,68 @@ protected function update_storage_config($storage_name)
 		}
 
 		// Update provider
-		$this->config->set('storage\\' . $storage_name . '\\provider', $this->get_new_provider($storage_name));
+		$this->config->set('storage\\' . $storage_name . '\\provider', $this->state['storages'][$storage_name]['provider']);
 
 		// Set new storage config
-		$new_options = $this->get_provider_options($this->get_new_provider($storage_name));
+		$new_options = $this->get_provider_options($this->state['storages'][$storage_name]['provider']);
 
 		foreach (array_keys($new_options) as $definition)
 		{
-			$this->config->set('storage\\' . $storage_name . '\\config\\' . $definition, $this->get_new_definition($storage_name, $definition));
+			$this->config->set('storage\\' . $storage_name . '\\config\\' . $definition, $this->state['storages'][$storage_name]['config'][$definition]);
 		}
 	}
+
+	protected function get_current_adapter($storage_name)
+	{
+		static $adapters = [];
+
+		if(!isset($adapters[$storage_name]))
+		{
+			$provider = $this->get_current_provider($storage_name);
+			$provider_class = $this->provider_collection->get_by_class($provider);
+
+			$adapter = $this->adapter_collection->get_by_class($provider_class->get_adapter_class());
+			$definitions = $this->get_provider_options($provider);
+
+			$options = [];
+			foreach (array_keys($definitions) as $definition)
+			{
+				$options[$definition] = $this->get_current_definition($storage_name, $definition);
+			}
+
+			$adapter->configure($options);
+			//$adapter->set_storage($storage_name);
```

Comment:
commented out code

Reply:
This have to be uncommented on other pr, but is right